doc/source/reference/index.rst

```diff
@@ -41,25 +41,26 @@ to import modules/functions/objects from SciPy.
 Guidelines for importing functions from SciPy
 ---------------------------------------------
 
-The scipy namespace itself only contains functions imported from numpy.  These
+The SciPy namespace itself only contains functions imported from NumPy. These
```

Comment:
I'm not sure this opening paragraph is strictly true anymore, at least from the user's perspective as defined by `__dir__`:

```python
>>> >>> import scipy
>>> dir(scipy)
['LowLevelCallable', '__numpy_version__', '__version__', 'cluster', 'datasets', 'fft', 'fftpack', 'integrate', 'interpolate', 'io', 'linalg', 'misc', 'ndimage', 'odr', 'optimize', 'show_config', 'signal', 'sparse', 'spatial', 'special', 'stats', 'test']
>>> scipy.__version__
'1.10.1'
```

Reply:
So it's a bit weird but we sadly still have some NumPy stuff 😓 Like

```python
import scipy as sp
sp.rand
# <function RandomState.rand>
```

We do raise a warning now if you try to use these. I am happy to remove the mention at all. But some folks here prefer that we describe exactly the current state of things vs what we want people to know.

Reply:
Gotcha - this is just one area where the module `__dir__` and `__getattr__` muddy the waters about what "public" means.

Reply:
On the actual diff: "the scipy namespace" refers to a code object (a namespace), so lower-case seems right. Maybe add backticks instead to typeset as code?

Reply:
> But some folks here prefer that we describe exactly the current state of things vs what we want people to know.

Unfortunately, this is neither. Because of the lazy import mechanism, the namespace now includes the subpackages and does not "only contain the functions imported from `numpy`". To document the current state of things _and_ what we want people to know, we should elevate the lazy import mechanism up to the top here and relegate the mention that there are `numpy` functions down to the `.. note::`

Reply:
You could just move this paragraph down to the end under a `.. note::` and remove "only".